go-controller/pkg/metrics/metrics.go

```diff
@@ -144,6 +172,177 @@ func coverageShowMetricsUpdater(component string) {
 	}
 }
 
+// registerStopwatchShowMetrics registers stopwatch/show metrics for
+// various components(ovn-northd, ovn-controller) with prometheus
+func registerStopwatchShowMetrics(component string, metricNamespace string, metricSubsystem string) {
+	stopwatchShowMetricsMap := componentStopwatchShowMetricsMap[component]
+	for metricName, metricInfo := range stopwatchShowMetricsMap {
+		metricInfo.metrics.totalSamples = prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Subsystem: metricSubsystem,
+			Name:      fmt.Sprintf("%s_total_samples", metricName),
+		})
+
+		metricInfo.metrics.max = prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Subsystem: metricSubsystem,
+			Name:      fmt.Sprintf("%s_maximum", metricName),
+		})
+
+		metricInfo.metrics.min = prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Subsystem: metricSubsystem,
+			Name:      fmt.Sprintf("%s_minimum", metricName),
+		})
+
+		metricInfo.metrics.percentile95th = prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Subsystem: metricSubsystem,
+			Name:      fmt.Sprintf("%s_95th_percentile", metricName),
+		})
+
+		metricInfo.metrics.shortTermAvg = prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Subsystem: metricSubsystem,
+			Name:      fmt.Sprintf("%s_short_term_avg", metricName),
+		})
+
+		metricInfo.metrics.longTermAvg = prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Subsystem: metricSubsystem,
+			Name:      fmt.Sprintf("%s_long_term_avg", metricName),
+		})
+
+		prometheus.MustRegister(metricInfo.metrics.totalSamples)
+		prometheus.MustRegister(metricInfo.metrics.min)
+		prometheus.MustRegister(metricInfo.metrics.max)
+		prometheus.MustRegister(metricInfo.metrics.percentile95th)
+		prometheus.MustRegister(metricInfo.metrics.shortTermAvg)
+		prometheus.MustRegister(metricInfo.metrics.longTermAvg)
+	}
+}
+
+// getStopwatchShowOutputMap obtains the stopwatch/show metric values for the specified component.
+func getStopwatchShowOutputMap(component string) (map[string]stopwatchStatistics, error) {
+	var stdout, stderr string
+	var err error
+
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("recovering from a panic while parsing the stopwatch/show output for %s: %v", component, r)
+		}
+	}()
```

Comment:
What's the reasoning behind having this here? We might as well be recovering all over the code in most functions, what makes this special?

Reply:
If you end up removing this function: then the function should also finish by `return parsedOutput, nil`

Reply:
As discussed, I checked the `util.RunOVNControllerAppCtl` and `util.RunOVNNorthAppCtl` function and since both are using the `afero.ReadFile` function, they can panic as well. And these panics are not directly under our control (e.g. if the output of stopwatch/show from ovn-northd is too big that it could be captured). I think it's only a corner case, but could be prevented by the recover.